addons/website_slides/models/slide_slide.py

```diff
@@ -227,15 +235,18 @@ def create(self, values):
             doc_data = self._parse_document_url(values['url']).get('values', dict())
             for key, value in doc_data.items():
                 values.setdefault(key, value)
-        # Do not publish slide if user has not publisher rights
-        if not self.user_has_groups('website.group_website_publisher'):
-            values['website_published'] = False
+
         slide = super(Slide, self).create(values)
-        slide._post_publication()
+
+        if slide.website_published:
+            slide._post_publication()
         return slide
 
     @api.multi
     def write(self, values):
+        if values.get('website_published') and any(not slide.channel_id.can_publish for slide in self):
```

Comment:
In the "create" method you also handle a "date_published", ne need to handle it here?

Reply:
Well it is to ensure someone does not try to publish its slide via some custom rpc / direct call, even if normally access rights prevent it.